src/extension/background-script/db.ts

```diff
@@ -14,7 +14,7 @@ class DB extends Dexie {
       allowances:
         "++id,&host,name,imageURL,tag,enabled,totalBudget,remainingBudget,lastPaymentAt,lnurlAuth,createdAt",
       payments:
-        "++id,allowanceId,host,location,name,description,totalAmount,totalFees,preimage,paymentRequest,paymentHash,destination,createdAt",
+        "++id,allowanceId,host,location,name,description,totalAmount,totalFees,preimage,paymentRequest,metadata,paymentHash,destination,createdAt",
```

Comment:
now we store metadata as a string. I was thinking if we store it as JSON then it should be easier to search for certain things. 

maybe it is also useful to use a new table for this and then split up the metadata into different columns? do you have thoughts on this?

Reply:
Sure! I can experiment and make changes when we decide how and what content we render in the transaction list (already there is PR representing rendering of metadata), in whichever cases we need to query metadata.


Now I am thinking of puttin metadata in a format of JSON as object

IndexedDB stores javascript objects. So if your documents are JSON docs, they can be perfectly stored as objects (such as the result from for example JSON.parse(json)). If you choose to store the JSON as text instead, that would work as well, but in case you like to index certain properties, you must store them as objects.


JSON as a blob has few drawback according to the stackoverflow post

```
Putting a JSON blob in a column makes it virtually impossible to query (and painfully slow when you actually find a query that works)
```

Reference::

https://github.com/dexie/Dexie.js/issues/823
https://stackoverflow.com/questions/15367696/storing-json-in-database-vs-having-a-new-column-for-each-key


regarding making a relational table and storing metadata as one attribute per column might have its own complexities but sure until I try! but as dexie can store javascript objects, this might not needed if query is simple to extract fields from the metadata object